### PR TITLE
Refactoring specs to DRY them up a bit

### PR DESCRIPTION
**Pull request by petems**
Been reading http://betterspecs.org/ recently and I thought I could implement a few ideas...

Just opening a PR to see if Travis doesn't barf, then I'll merge...

### CODE REVIEW

File: spec/capistrano-spec_spec.rb [hunk 1]
```diff
@@ -19,15 +21,19 @@
     end
 
     it "has a #callback matcher" do
-      expect{@configuration.should callback(foo)}.to_not raise_error(NoMethodError)
+      expect{ should callback(foo)}.to_not raise_error(NoMethodError)
```
Comment (technicalpickles) [inline spec/capistrano-spec_spec.rb]: Aren't all these should calls extraneous? At the very least, they read very weirdly.

**Comment by relistan** [comment]
Sorry, been on vacation for a couple of weeks and missed a lot of action it seems.  Can't see why the original tests were calling both expect and should in a number of places, either.  Normally you use one syntax or the other.  `expect { something }.to` is essentially equivalent to `lambda { something }.should` in RSpec 1.

EDIT: Ah, OK, the calls that are actually being tested _are_ the `should foo` calls as these are the matchers in this gem.  But, actually looking at this more closely, a lot of those expects are not needed.  If executing the call raises an error it will just cause the test to fail anyway.  The `should` calls are the ones that should be retained in most of the cases, except when specific exception checking is needed.  Maybe I missed something about what is being checked.  But I would think for example that this:

``` ruby
    it "will not raise error when run is in recipe" do
      @configuration.find_and_execute_task('fake:thing')
      expect { @configuration.should have_run("do some stuff")}.to_not raise_error(RSpec::Expectations::ExpectationNotMetError, /expected configuration to run .*\s*, but did not/)
    end
```

Should really just be:

``` ruby
    it "runs properly in a recipe" do
      @configuration.find_and_execute_task('fake:thing')
      @configuration.should have_run("do some stuff")
    end
```

File: spec/capistrano-spec_spec.rb (hunk 1 above)
Comment (petems) [inline spec/capistrano-spec_spec.rb]: Yes probably, I'll remove these.

**Comment by petems** [comment]
@relistan Yeah, I think I went a bit overkill with the expects, it's a bit weird testing tests with tests...

I've tweaked it up a bit now, thoughts?

File: spec/capistrano-spec_spec.rb (hunk 1 above)
Comment (petems) [inline spec/capistrano-spec_spec.rb]: Done in https://github.com/technicalpickles/capistrano-spec/commit/94fa8e35e72db8e23704271e962735a59b011b4e

**Comment by petems** [comment]
@relistan It turns out that if I don't have those `should_not`'s in there, and I remove the corresponding stuff from the recipe, the tests are still green, so I've put them back in.

**Pull request merged**